lexer/lexer.cc

```diff
@@ -210,18 +210,131 @@ Token Lexer::parseStringLiteral() {
   // TODO handle raw string/heredoc
   const char *firstPos = mLastPosition;
   const char *lastPos = mLastPosition;
+
   while (mLast != '"') {
-    if (mLast == 0)
+    if (mLast == 0) {
       throw std::runtime_error(fmt::format(
-          "{}: Lexer error: unerminated string literal", mLocation));
+          "{}: Lexer error: unterminated string literal", mLocation));
+    }
+
+    if (mLast == '\\') {
+      // Got escaped character. Handle this string literal separately.
+      return parseStringLiteralWithEscapedCharacters(firstPos);
+    }
+
     readNextChar();
+
     lastPos = mLastPosition;
   }
+
   readNextChar(); // skip past the closing quote
   std::string_view content(firstPos, lastPos - firstPos);
   return Token{TokenType::LitString, mTokenBeginLocation, content};
 }
 
+Token Lexer::parseStringLiteralWithEscapedCharacters(const char *firstPos) {
+
+  std::string result(firstPos, mLastPosition);
+
+  // At this point the '\' of the first escaped character has been already read.
+
+  while (mLast != '"') {
+    if (mLast == 0) {
+      throw std::runtime_error(fmt::format(
+          "{}: Lexer error: unterminated string literal", mLocation));
+    }
+
+    if (mLast == '\\') {
+      // Got escaped character. Handle this string literal separately.
+      result += readEscapedCharacter();
+    } else {
+      result += mLast;
+    }
+
+    readNextChar();
+  }
+
+  readNextChar(); // skip past the closing quote
+  std::string_view content = Utils::Zone::strdup(result);
+  return Token{TokenType::LitString, mTokenBeginLocation, content};
+}
+
+Utils::rune_t Lexer::readEscapedCharacter() {
+  readNextChar();
+  if (mLast == 0) {
+    throw std::runtime_error(
```

Comment:
I think we need a special exception type, that holds both message and location separately, so that the errors can be processed in systematic way and integrates into diagnostics system. (this is just an idea  for the future)